command/whitelist/deployment/params.go

```diff
@@ -0,0 +1,166 @@
+	"github.com/0xPolygon/polygon-edge/command"
+	"github.com/0xPolygon/polygon-edge/command/helper"
+	"github.com/0xPolygon/polygon-edge/helper/common"
+	"github.com/0xPolygon/polygon-edge/types"
+)
+
+const (
+	chainFlag         = "chain"
+	addAddressFlag    = "addAddress"
+	removeAddressFlag = "removeAddress"
+)
+
+var (
+	errInvalidAddressFormat = errors.New("one or more addresses are of invalid format")
+)
+
+var (
+	params = &deploymentParams{}
+)
+
+type deploymentParams struct {
+	// raw addresses, entered by CLI commands
+	addAddressRaw    []string
+	removeAddressRaw []string
+
+	// addresses, converted from raw addresses
+	addAddress    []types.Address
+	removeAddress []types.Address
+
+	// genesis file
+	genesisPath   string
+	genesisConfig *chain.Chain
+
+	// deployment whitelist from genesis configuration
+	whitelist []types.Address
+}
+
+func (p *deploymentParams) initRawParams() error {
+	// convert raw addresses to appropriate format
+	if err := p.initRawAddresses(); err != nil {
+		return err
+	}
+
+	// init genesis configuration
+	if err := p.initChain(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (p *deploymentParams) initRawAddresses() error {
+	// init slices for addresses to be added and removed
+	p.addAddress = []types.Address{}
+	p.removeAddress = []types.Address{}
+
+	// convert addresses to be added from string to type.Address
+	for _, address := range p.addAddressRaw {
+		newAddress := types.Address{}
+
+		if err := newAddress.UnmarshalText([]byte(address)); err != nil {
+			return errInvalidAddressFormat
+		}
+
+		p.addAddress = append(p.addAddress, newAddress)
+	}
+
+	// convert addresses to be removed from string to type.Address
+	for _, address := range p.removeAddressRaw {
+		newAddress := types.Address{}
+
+		if err := newAddress.UnmarshalText([]byte(address)); err != nil {
+			return errInvalidAddressFormat
+		}
+
+		p.removeAddress = append(p.removeAddress, newAddress)
+	}
+
+	return nil
+}
+
+func (p *deploymentParams) initChain() error {
+	// import genesis configuration
+	cc, err := chain.Import(p.genesisPath)
+	if err != nil {
+		return fmt.Errorf(
+			"failed to load chain config from %s: %w",
+			p.genesisPath,
+			err,
+		)
+	}
+
+	// set genesis configuration
+	p.genesisConfig = cc
+
+	return nil
+}
+
+func (p *deploymentParams) updateGenesisConfig() error {
+	// Fetch contract deployment whitelist from genesis config
+	deploymentWhitelist, err := common.FetchDeploymentWhitelist(p.genesisConfig)
+	if err != nil {
+		return err
+	}
+
+	// Add addresses if not exists
+	for _, address := range p.addAddress {
+		if !address.ExistsIn(deploymentWhitelist) {
+			deploymentWhitelist = append(deploymentWhitelist, address)
+		}
+	}
+
+	newDeploymentWhitelist := make([]types.Address, 0)
+
+	// Remove addresses if exists
+	for _, address := range deploymentWhitelist {
+		if !address.ExistsIn(p.removeAddress) {
+			newDeploymentWhitelist = append(newDeploymentWhitelist, address)
+		}
+	}
```

Comment:
Why do you need to define a new array here?
Why not just work with `deploymentWhitelist`?

This should also be a helper

Reply:
Because I don't like doing the removal, I would need to find the index and then to remove that index from slice, i prefer creating a new array and just populate it with the items which are not removed

Reply:
See my other comment

Reply:
Fixed 2ea2ad34a752974a637e7e4562eb8c36cff33678